code/model/VirtualPage.php

```diff
@@ -413,7 +413,7 @@ public function CMSTreeClasses($numChildrenMethod="numChildren") {
 	public function __get($field) {
 		if(parent::hasMethod($funcName = "get$field")) {
 			return $this->$funcName();
-		} else if(parent::hasField($field)) {
+		} else if(parent::hasField($field) || ($field === 'ID' && !$this->getField('ID'))) {
```

Comment:
`!$this->exists()` is a nicer way of checking the ID is non-zero and being more self documenting